src/subset.h

```diff
@@ -0,0 +1,30 @@
+#ifndef SUBSET_H
```

Comment:
Careful with the namespacing here: pick one name prefix and use it for everything in the file.  It may be obvious to you now, but will get confusing for others later...